### PR TITLE
API: if config is not found, don't import DataBroker

### PR DESCRIPTION
**Pull request by chiahaoliu**
Author: @danielballan

### CODE REVIEW

**Comment by danielballan** [comment]
As we move toward explicitly defining MDS and FS objects in the beamline configs, do the following on `import databroker`:

If MDS or FS raise because they can't find global config, warn, and do not import the top-level funcs (get_events, get_table, DataBroker). It's *very* unusual to have contingent imports but I think alternatives (require global config forever; replace `DataBroker` with `None`...) are worse.

The whole issue is that `DataBroker`, `get_events`, and friends are importable from the top level `__init__` and that import is *very* ingrained with staff at this point. For this reason, `from databroker import db` should continue to work forever for users that have a global config in place.

**Comment by danielballan** [comment]
bump @tacaswell @arkilic

File: databroker/__init__.py [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
+    from .pims_readers import get_images
+    from .handler_registration import register_builtin_handlers
+
+    # register all built-in filestore handlers
+    register_builtin_handlers()
+    del register_builtin_handlers
+except ImportError:
+    warnings.warn("The top-level functions (get_table, get_events, etc.)"
+                  "cannot be created because "
+                  "the necessary configuration was not found.")
```
Comment (klauer) [inline databroker/__init__.py]: Might want to include the exception message here as this is a big try block

**Comment by danielballan** [comment]
Superceded by #66

**Comment by danielballan** [comment]
Err, #75

File: databroker/__init__.py (hunk 1 above)
Comment (danielballan) [inline databroker/__init__.py]: Addressed in #75. Thanks.

**Pull request closed without merging**